pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```diff
@@ -47,38 +70,54 @@ let
         ${replaced} = v;
         "tree-sitter-${replaced}" = v;
       }
-    ) generatedDerivations;
+    ) parsersWithMeta;
 
-  allGrammars = lib.attrValues generatedDerivations;
+  allGrammars = builtins.attrValues parsersWithMeta;
 
   # Usage:
   # pkgs.vimPlugins.nvim-treesitter.withPlugins (p: [ p.c p.java ... ])
   # or for all grammars:
   # pkgs.vimPlugins.nvim-treesitter.withAllGrammars
   withPlugins =
     f:
+    let
+      selectedGrammars = f (tree-sitter.builtGrammars // builtGrammars);
+
+      grammarPlugins = map grammarToPlugin selectedGrammars;
+
+      queryPlugins = lib.pipe selectedGrammars [
+        (map (g: g.passthru.associatedQuery or null))
+        (builtins.filter (q: q != null))
+      ];
+    in
     self.nvim-treesitter.overrideAttrs {
-      passthru.dependencies = map grammarToPlugin (f (tree-sitter.builtGrammars // builtGrammars));
+      passthru.dependencies = [
+        (symlinkJoin {
+          name = "nvim-treesitter-grammars";
+          paths = grammarPlugins ++ queryPlugins;
+        })
```

Comment:
This symlinks all grammars into one derivation; even if now nvim-treesitter supports this, it must be done in the wrapper, and not `withPlugins`. Currently, this will definitely lead to duplicate grammars based on plugin dependencies, so we should definitely not shoot ourselves in the leg *again*

Reply:
> I also think doing it in the wrapper is probably better, but I'm not sure if it should block this PR.

I plan to make it for the wrapper, just added in it in the `withPlugins` as a stand-in. As we have people that don't use the nixpkgs neovim wrapper, and we didn't deprecate `withPlugins` yet